std/http/server.ts

```diff
@@ -180,24 +178,6 @@ export class Server implements AsyncIterable<ServerRequest> {
       await req.finalize();
     }
 
-    if (req === Deno.EOF) {
-      // The connection was gracefully closed.
-    } else if (err && req) {
-      // An error was thrown while parsing request headers.
-      try {
-        await writeResponse(req.w, {
-          status: 400,
-          body: encode(`${err.message}\r\n\r\n`)
-        });
-      } catch (_) {
```

Comment:
See #3992. The request this attempts to write to has already been yielded and it's `done` has already been awaited.